commons-numbers-gamma/src/main/java/org/apache/commons/numbers/gamma/LogBeta.java

```diff
@@ -0,0 +1,290 @@
+     * @throws IllegalArgumentException if {@code a < 0} or {@code a > b}
+     * @throws IllegalArgumentException if {@code b < 10}
+     */
+    private static double deltaMinusDeltaSum(final double a,
+                                             final double b) {
+        if (a < 0 ||
+            a > b) {
+            throw new GammaException(GammaException.OUT_OF_RANGE, a, 0, b);
+        }
+        if (b < TEN) {
+            throw new GammaException(GammaException.OUT_OF_RANGE, b, TEN, Double.POSITIVE_INFINITY);
+        }
+
+        final double h = a / b;
+        final double p = h / (1 + h);
+        final double q = 1 / (1 + h);
+        final double q2 = q * q;
+        /*
+         * s[i] = 1 + q + ... - q**(2 * i)
+         */
+        final double[] s = new double[DELTA.length];
+        s[0] = 1;
+        for (int i = 1; i < s.length; i++) {
+            s[i] = 1 + (q + q2 * s[i - 1]);
+        }
+        /*
+         * w = Delta(b) - Delta(a + b)
+         */
+        final double sqrtT = 10 / b;
+        final double t = sqrtT * sqrtT;
+        double w = DELTA[DELTA.length - 1] * s[s.length - 1];
+        for (int i = DELTA.length - 2; i >= 0; i--) {
+            w = t * w + DELTA[i] * s[i];
+        }
+        return w * p / b;
+    }
+
+    /**
+     * Returns the value of \( \Delta(p) + \Delta(q) - \Delta(p + q) \),
+     * with \( p, q \geq 10 \).
+     * Based on the <em>NSWC Library of Mathematics Subroutines</em> implementation,
+     * {@code DBCORR}.
+     *
+     * @param p First argument.
+     * @param q Second argument.
+     * @return the value of \( \Delta(p) + \Delta(q) - \Delta(p + q) \).
+     * @throws IllegalArgumentException if {@code p < 10} or {@code q < 10}.
+     */
+    private static double sumDeltaMinusDeltaSum(final double p,
+                                                final double q) {
+
+        if (p < TEN) {
+            throw new GammaException(GammaException.OUT_OF_RANGE, p, TEN, Double.POSITIVE_INFINITY);
+        }
+        if (q < TEN) {
+            throw new GammaException(GammaException.OUT_OF_RANGE, q, TEN, Double.POSITIVE_INFINITY);
+        }
+
+        final double a = Math.min(p, q);
+        final double b = Math.max(p, q);
+        final double sqrtT = 10 / a;
+        final double t = sqrtT * sqrtT;
+        double z = DELTA[DELTA.length - 1];
+        for (int i = DELTA.length - 2; i >= 0; i--) {
+            z = t * z + DELTA[i];
+        }
+        return z / a + deltaMinusDeltaSum(a, b);
+    }
+
+    /**
+     * Returns the value of \( \log B(p, q) \) for \( 0 \leq x \leq 1 \) and \( p, q &gt; 0 \).
+     * Based on the <em>NSWC Library of Mathematics Subroutines</em> implementation,
+     * {@code DBETLN}.
+     *
+     * @param p First argument.
+     * @param q Second argument.
+     * @return the value of \( \log B(p, q) \), or {@code NaN} if
+     * {@code p <= 0} or {@code q <= 0}.
+     */
+    public static double value(double p,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.